R/parse_expr.R

```diff
@@ -271,7 +271,7 @@ parse_expr_assignment_rhs_dim <- function(rhs, src, call) {
       "E1040", src, call)
   }
   ## See parse_expr_check_lhs_index, which has similar restrictions
-  allowed <- c("+", "-", "(", "length", "nrow", "ncol")
+  allowed <- c("+", "-", "(", "length", "nrow", "ncol", "dim")
```

Comment:
I think it could be simpler if we special cased this right at the start of this function rather than coping with it within the general parse, so

```
if (rlang::is_call(rhs, "dim")) {
  ## check that arg is a symbol only
  ## early return as
  list(type = "dim", is_user_sized = FALSE, is_dependent_dim = TRUE, value = rhs, depends = as.character(rhs[[2]]))
}
...existing impl
```